src/main/java/seedu/address/model/person/NextOfKin.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Patient's Next of Kin in the Patient Database.
+ * Guarantees: immutable; is valid as declared in {@link #isValidNextOfKin(String)}
+ */
+public class NextOfKin {
+
+    public static final String MESSAGE_CONSTRAINTS = "Next of Kin should not be blank "
+            + "and should come in the format: Name, Relationship, Contact";
+
+    public static final String VALIDATION_REGEX = "[A-Za-z\\s]+,[a-zA-Z\\s]+,[0-9\\s]+";
```

Comment:
Validation regex looks good! 
 ** For parser regex for get /nok, will need to consider the case when name has spaces in between (First and last name).

Reply:
Agree, for now, it only checks that the three fields are there, no validation is done for the actual fields, only that they must have at least 1 alpha/numeric or space character.